### PR TITLE
feat: add message mention highlighting (everyone + user)

### PR DESCRIPTION
**Pull request by DiamondMiner88**
- Add message mention highlighting
- Role mentions are currently left out because there's no role model

image

### CODE REVIEW

File: app/src/main/java/com/xinto/opencord/ui/viewmodel/ChatViewModel.kt [hunk 1]
```diff
@@ -42,6 +43,8 @@ class ChatViewModel(
         private set
     var sendEnabled by mutableStateOf(true)
         private set
+    var currentUserId by mutableStateOf<Long?>(null)
```
Comment (X1nto) [inline app/src/main/java/com/xinto/opencord/ui/viewmodel/ChatViewModel.kt:46]: Could be 0 instead of null

Comment (DiamondMiner88) [reply app/src/main/java/com/xinto/opencord/ui/viewmodel/ChatViewModel.kt:46]: Why though, its set from ReadyEvent so it'll always be missing or valid. null exists to represent absense of data

Comment (ItzOnlyAnimal) [reply app/src/main/java/com/xinto/opencord/ui/viewmodel/ChatViewModel.kt:46]: I guess so if ready event doesn't fire for whatever reason

Comment (DiamondMiner88) [reply app/src/main/java/com/xinto/opencord/ui/viewmodel/ChatViewModel.kt:46]: Therefore null?

Comment (X1nto) [reply app/src/main/java/com/xinto/opencord/ui/viewmodel/ChatViewModel.kt:46]: Okay yeah sure

**Review by X1nto — APPROVED** [review]

**Pull request merged**